### PR TITLE
[#173351990] fromQueryEither

### PR DESCRIPTION
**Pull request by balanza**
## Motivation
Our db interfaces return results in the unpleasant form `Promise<Either<QueryError, T>>` while a form `TaskEither<QueryError, T>` would be preferrable. `fromQueryEither` is a helper method that does such type conversion.
This helper [has been used extensively in our codebase](https://github.com/search?q=fromQueryEither+user%3Apagopa&type=Code&ref=advsearch&l=&l=) and it needs to be defined as a common util to avoid duplication.

## Implementation
- Cosmos' query errors are returned as left 
- Runtime errors (that is: when the promise rejects for unexpected problems) are mapped in the form `{code: "error", body: error.message}` to mock the interface of Cosmos' QueryError type
- a convenient `DocumentDbUtils.QueryError` type is defined to map such union 

## Usage
see [test snippets](https://github.com/pagopa/io-functions-commons/blob/1beb3a112c55111f7633505aeec41326df863fa2/src/utils/__tests__/documentdb.test.ts#L676).

### CODE REVIEW

File: src/utils/__tests__/documentdb.test.ts [hunk 1]
```diff
@@ -671,3 +672,71 @@ describe("queryAttachments", () => {
...
+  });
+
+  it("should return the correct query error on rejection", async () => {
+    interface IExpectedResultType {
+      foo: string;
+    }
+
+    const errorMessage = "generic error";
+    const queryFn: () => Promise<
+      Either<DocumentDb.QueryError | Error, IExpectedResultType>
```
Comment (gunzip) [inline src/utils/__tests__/documentdb.test.ts:727]: It looks like this promise here (the queryfn) cannot never return `left(Error)`

```suggestion
      Either<DocumentDb.QueryError, IExpectedResultType>
```

File: src/utils/documentdb.ts [hunk 2]
```diff
@@ -790,4 +791,31 @@ export function queryAttachments<T>(
...
+
+/**
+ * Converts a Promise<Either> into a TaskEither
+ * This is needed because our models return unconvenient type. Both left and rejection cases are handled as a TaskEither left
+ * @param lazyPromise a lazy promise to convert
+ *
+ * @returns either the query result or a query failure
+ */
+export const fromQueryEither = <R>(
+  lazyPromise: () => Promise<Either<DocumentDb.QueryError | Error, R>>
```
Comment (gunzip) [inline src/utils/documentdb.ts:809]: Why the signature of this promise is 

`Promise<Either<DocumentDb.QueryError | Error, R>>` and not just

`Promise<Either<DocumentDb.QueryError, R>>` ?

Comment (balanza) [reply src/utils/documentdb.ts:809]: For a more relaxed API, but I think we can be more specific

File: src/utils/documentdb.ts [hunk 3]
```diff
@@ -790,4 +791,31 @@ export function queryAttachments<T>(
...
 
+/**
+ * A convenient type that extends Cosmos' DocumentDb.QueryError with a generic status that maps a runtime error (instance of Error)
+ */
+export type QueryError =
+  | DocumentDb.QueryError
+  | { code: "error"; body: string };
+
+/**
+ * Converts a Promise<Either> into a TaskEither
```
Comment (cloudify) [inline src/utils/documentdb.ts:802]:
```suggestion
 * Converts a Promise<Either<L, R>> into a TaskEither<L, R>
```

**Pull request merged**